hazelcast-sql/src/test/java/com/hazelcast/jet/sql_nightly/SqlSTSInnerEquiJoinFaultToleranceStressTest.java

```diff
@@ -0,0 +1,287 @@
+        kafkaTestSupport.shutdownKafkaCluster();
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        sqlService = instance().getSql();
+
+        // Kafka source definition
+        sourceTopicName = createRandomTopic();
+        sqlService.execute("CREATE MAPPING " + sourceTopicName + ' '
+                + "TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
+                + "OPTIONS ( "
+                + '\'' + OPTION_KEY_FORMAT + "'='int'"
+                + ", '" + OPTION_VALUE_FORMAT + "'='varchar'"
+                + ", 'bootstrap.servers'='" + kafkaTestSupport.getBrokerConnectionString() + '\''
+                + ", 'auto.offset.reset'='earliest'"
+                + ")");
+
+        kafkaFeedThread = new Thread(() -> createTopicData(sqlService, sourceTopicName));
+        kafkaFeedThread.start();
+
+        // Kafka sink topic definition
+        sinkTopic = randomName();
+        sqlService.execute("CREATE MAPPING " + sinkTopic
+                + " TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
+                + " OPTIONS ( "
+                + '\'' + OPTION_KEY_FORMAT + "'='int'"
+                + ", '" + OPTION_VALUE_FORMAT + "'='varchar'"
+                + ", 'bootstrap.servers'='" + kafkaTestSupport.getBrokerConnectionString() + '\''
+                + ", 'auto.offset.reset'='earliest'"
+                + ")");
+
+        // Left & right sides of the JOIN
+        sqlService.execute("CREATE VIEW s1 AS " +
+                "SELECT * FROM TABLE(IMPOSE_ORDER(TABLE " + sourceTopicName + " , DESCRIPTOR(__key), 10))");
+        sqlService.execute("CREATE VIEW s2 AS " +
+                "SELECT * FROM TABLE(IMPOSE_ORDER(TABLE " + sourceTopicName + " , DESCRIPTOR(__key), 5))");
+
+        fetchingQuery = setupFetchingQuery();
+
+        Thread.sleep(15_000L); // time to feed Kafka
+    }
+
+    @Test
+    public void stressTest() {
+        JobRestarter jobRestarter = new JobRestarter(instance());
+        jobRestarter.start();
+
+        processAndCheckResults(jobRestarter);
+
+        try {
+            kafkaFeedThread.join();
+            jobRestarter.join();
+        } catch (AssertionError e) {
+            Assert.fail(e.getMessage());
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected void processAndCheckResults(JobRestarter jobRestarter) {
+        Thread thread = new Thread(() -> {
+            SqlStatement statement = new SqlStatement(fetchingQuery);
+
+            try (SqlResult result = sqlService.execute(statement)) {
+            } catch (Throwable e) {
+                e.printStackTrace();
+            }
+        });
+        thread.start();
+
+        String mainCheckQuery = "SELECT * FROM " + sinkTopic;
+        try (SqlResult result = sqlService.execute(mainCheckQuery)) {
+            for (SqlRow sqlRow : result) {
+                String s = sqlRow.getObject(1);
+                resultSet.compute(s, (str, i) -> i == null ? 0 : i + 1);
```

Comment:
Why do you start with 0? Below you fail if value is `> 1`, that is if the row was twice in the result. Mistake?

What about this:
```suggestion
                resultSet.merge(s, 1, Integer::sum);
```